datagateway_api/src/search_api/query_filter_factory.py

```diff
@@ -94,40 +124,17 @@ def get_include_filter(filter_input):
             query_filters.append(SearchAPIIncludeFilter(included_entity))
 
             if "scope" in related_model:
-                try:
-                    where_key = list(related_model["scope"]["where"].keys(),)[0]
-
-                    if where_key == "text":
-                        # TODO - we might want to move this to the data
-                        # definitions at a later point
-                        text_operator_fields = {
-                            "datasets": ["title"],
-                            "documents": ["title", "summary"],
-                            "files": ["name"],
-                            "instrument": ["name", "facility"],
-                            "samples": ["name", "description"],
-                            "techniques": ["name"],
-                        }
-                        field_names = text_operator_fields[included_entity]
-                    else:
-                        field_names = [where_key]
-
-                    for field_name in field_names:
-                        full_field_path = f"{included_entity}.{field_name}"
-                        where_filter = {
-                            "filter": {
-                                "where": {
-                                    full_field_path: related_model["scope"]["where"][
-                                        where_key
-                                    ],
-                                },
-                            },
-                        }
-                        query_filters.extend(
-                            SearchAPIQueryFilterFactory.get_query_filter(where_filter,),
+                # Scope filter can have WHERE, INCLUDE, LIMIT and SKIP filters
```

Comment:
Are there examples of there being include, limit and skip filters inside the scope property? Or do they work in the example implementation? I assumed only a WHERE is expected inside the scope, but maybe all the filters work?

Reply:
Not in the _Querying_ mark down but I did find a few examples of the include filter being inside the scope property in the [Document queries](https://github.com/panosc-eu/search-api/blob/master/doc/document-example-queries.md) mark down. This prompted me into looking at the [Loopback 4 Node.js](https://loopback.io/doc/en/lb4/Include-filter.html#include-with-filters) documentation where I discovered that scope supports limit and skip filters in addition to where and include. While looking for the LoopBack link just now, I also noticed that fields is another filter listed in there. I can confirm that all of these filters work in the example Search API implementation.

Reply:
Good spot, I hadn't realised this. Does your change [here](https://github.com/ral-facilities/datagateway-api/blob/cdb30ff0d4ebc8a0901e385d14bbd623f7c96fbb/datagateway_api/src/search_api/query_filter_factory.py#L128) mean we support all filters in the `scope` property now? Sadly this means adding tests that they work... I know `test_valid_include_filter()` is getting quite long now (well all the pytest params) so it might be worth splitting that up between 'regular' include filters (i.e. just includes) and then another test function for all the scope stuff. I'll leave the decision to you, just an option if that function get a bit unweldy to maintain :)

Reply:
>Does your change [here](https://github.com/ral-facilities/datagateway-api/blob/cdb30ff0d4ebc8a0901e385d14bbd623f7c96fbb/datagateway_api/src/search_api/query_filter_factory.py#L128) mean we support all filters in the `scope` property now?

Correct. Essentially, it takes whatever is inside `scope`, creates a new filter request and passes it to `get_query_filter` to deal with it because `get_query_filter` already knows how to deal with WHERE, INCLUDE, LIMIT and SKIP filters.